src/caffe/test/test_net.cpp

```diff
@@ -2224,4 +2323,31 @@ TYPED_TEST(NetTest, TestReshape) {
   }
 }
 
+TYPED_TEST(NetTest, TestSkipPropagateDown) {
+  // check bottom_need_backward if skip_propagat_down is false
+  this->InitSkipPropNet(false);
+  for (int layer_id = 0; layer_id < this->net_->layers().size(); ++layer_id) {
+    if (this->net_->layer_names()[layer_id] == "loss") {
+      // access to bottom_need_backward coresponding to label's blob
+      bool need_back = this->net_->bottom_need_backward()[layer_id][1];
+      // if skip_propagate_down is false, the loss layer will try to
+      // backpropagate on labels
+      CHECK_EQ(need_back, true)
+        << "bottom_need_backward should be True";
+    }
+  }
+  // check bottom_need_backward if skip_propagat_down is true
+  this->InitSkipPropNet(true);
```

Comment:
This should also check that the `Net`'s `layer_need_backward_` (not sure if this is currently exposed; feel free to add a public accessor for it if not) is set to false for all layers below the loss (and the first loop should check that all `layer_need_backward`s are true, except for the data layer).